Applications/Utils/ModelPreparation/PartitionMesh/NodeWiseMeshPartitioner.cpp

```diff
@@ -0,0 +1,515 @@
+        for (std::size_t i = partition.number_of_base_nodes; i < end_id; i++)
+        {
+            _nodes_global_ids[partition.nodes[i]->getID()] =
+                node_global_id_offset;
+            node_global_id_offset++;
+        }
+    }
+}
+
+void NodeWiseMeshPartitioner::writeMETIS(const std::string& file_name)
+{
+    std::ofstream os(file_name.data(), std::ios::trunc);
+    if (!os.is_open())
+    {
+        OGS_FATAL("Error: cannot open file %s. It may not exist! ",
+                  file_name.data());
+    }
+    std::vector<MeshLib::Element*> const& elements = _mesh->getElements();
+    os << elements.size() << " \n";
+    for (const auto* elem : elements)
+    {
+        os << elem->getNodeIndex(0) + 1;
+        for (unsigned j = 1; j < elem->getNumberOfNodes(); j++)
+        {
+            os << " " << elem->getNodeIndex(j) + 1;
+        }
+        os << "\n";
+    }
+    os.flush();
+}
+
+NodeWiseMeshPartitioner::IntegerType
+NodeWiseMeshPartitioner::getNumberOfIntegerVariablesOfElements(
+    const std::vector<const MeshLib::Element*>& elements) const
+{
+    IntegerType nmb_element_idxs = 3 * elements.size();
+    for (const auto* elem : elements)
+    {
+        nmb_element_idxs += elem->getNumberOfNodes();
+    }
+    return nmb_element_idxs;
+}
+
+void NodeWiseMeshPartitioner::writeBinary(const std::string& file_name_base)
+{
+    const std::string npartitions_str = std::to_string(_npartitions);
+
+    // Output configuration data
+    std::string fname =
+        file_name_base + "_partitioned_msh_cfg" + npartitions_str + ".bin";
+    FILE* of_bin_cfg = fopen(fname.c_str(), "wb");
+
+    const IntegerType num_config_data = 14;
+    IntegerType config_data[num_config_data];
+    // node rank offset
+    config_data[10] = 0;
+    // element rank offset
+    config_data[11] = 0;
+    // ghost element rank offset
+    config_data[12] = 0;
+    // Reserved
+    config_data[13] = 0;
+    std::vector<IntegerType> num_elem_integers(_partitions.size());
+    std::vector<IntegerType> num_g_elem_integers(_partitions.size());
+    std::size_t loop_id = 0;
+    for (const auto& partition : _partitions)
+    {
+        config_data[0] = partition.nodes.size();
+        config_data[1] = partition.number_of_base_nodes;
+        config_data[2] = partition.regular_elements.size();
+        config_data[3] = partition.ghost_elements.size();
+        config_data[4] = partition.number_of_non_ghost_base_nodes;
+        config_data[5] = partition.number_of_non_ghost_nodes;
+        config_data[6] = _mesh->getNumberOfBaseNodes();
+        config_data[7] = _mesh->getNumberOfNodes();
+        config_data[8] =
+            getNumberOfIntegerVariablesOfElements(partition.regular_elements);
+        config_data[9] =
+            getNumberOfIntegerVariablesOfElements(partition.ghost_elements);
+
+        fwrite(config_data, 1, num_config_data * sizeof(IntegerType), of_bin_cfg);
```

Comment:
The `std::ofstream` directly supports writing binary. No need to fallback to the c-library.
```c++
config_file.write(static_cast<char*>(config_data), num_config_data * sizeof(IntegerType));
```